### PR TITLE
Get rid of trivial anonymous functions (part 2)

### PR DESCRIPTION
**Pull request by lgoettgens**
See https://github.com/oscar-system/Oscar.jl/pull/4126 for part 1 and the reasoning behind these changes.

### CODE REVIEW

File: src/Modules/ExteriorPowers/FreeModules.jl [hunk 1]
```diff
@@ -173,8 +173,8 @@ function koszul_duals(v::Vector{T}; cached::Bool=true) where {T<:FreeModElem}
   success, M, p = _is_exterior_power(F)
   n = rank(M)
   success || error("element must be an exterior product")
-  k = [findfirst(x->x==u, gens(F)) for u in v]
-  any(x->x===nothing, k) && error("elements must be generators of the module")
+  k = [findfirst(==(u), gens(F)) for u in v]
+  any(is_nothing, k) && error("elements must be generators of the module")
```
Comment (lgoettgens) [inline src/Modules/ExteriorPowers/FreeModules.jl]:
```suggestion
  any(isnothing, k) && error("elements must be generators of the module")
```

File: src/Rings/mpoly-graded.jl [hunk 2]
```diff
@@ -1453,7 +1453,7 @@ end
 #########################################
 function add_relshp(R, S, h)
   #this assumes that h is essentially a canonical map from R -> S
-  D = get_attribute!(() -> Dict{Any, Any}(), R, :relshp)::Dict{Any, Any}
+  D = get_attribute!(R, :relshp, Dict{Any, Any}())::Dict{Any, Any}
```
Comment (fingolfin) [inline src/Rings/mpoly-graded.jl:1456]: The new code always allocates a new dictionary even if it is not needed. But you can do this:
```suggestion
  D = get_attribute!(Dict{Any, Any}, R, :relshp)::Dict{Any, Any}
```

**Review by fingolfin — APPROVED** [review]

File: src/Rings/mpoly-graded.jl (hunk 2 above)
Comment (fingolfin) [reply src/Rings/mpoly-graded.jl:1456]: All tests just passed, this can be done later

Comment (lgoettgens) [reply src/Rings/mpoly-graded.jl:1456]: Oooohhhh, that's sooo smart 😅

**Pull request merged**